### PR TITLE
make tolerations configurable

### PR DESCRIPTION
**Pull request by jsanda**
Needed for https://github.com/k8ssandra/k8ssandra/issues/698.

### CODE REVIEW

**Review by jeffbanks — APPROVED** [review]
LGTM.  A couple of comments for your review.

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -134,6 +135,13 @@ func (in *ReaperList) DeepCopyObject() runtime.Object {
 func (in *ReaperSpec) DeepCopyInto(out *ReaperSpec) {
 	*out = *in
```
[the remark this answers is not in the document]
Comment (jsanda) [reply api/v1alpha1/zz_generated.deepcopy.go:136]: We don't need to modify anything in this file. It is generated code.

File: config/crd/bases/reaper.cassandra-reaper.io_reapers.yaml [hunk 2]
```diff
@@ -93,6 +93,46 @@ spec:
                 storageType:
                   type: string
               type: object
+            tolerations:
+              description: Tolerations applied to the Reaper pods
+              items:
+                description: The pod this Toleration is attached to tolerates any
```
[the remark this answers is not in the document]
Comment (jsanda) [reply config/crd/bases/reaper.cassandra-reaper.io_reapers.yaml:99]: That description comes from the Toleration docs which is part of the core k8s api.

File: config/crd/bases/reaper.cassandra-reaper.io_reapers.yaml [hunk 3]
```diff
@@ -93,6 +93,46 @@ spec:
...
+                      all taints of a particular category.
+                    type: string
+                  tolerationSeconds:
+                    description: TolerationSeconds represents the period of time the
+                      toleration (which must be of effect NoExecute, otherwise this
+                      field is ignored) tolerates the taint. By default, it is not
+                      set, which means tolerate the taint forever (do not evict).
+                      Zero and negative values will be treated as 0 (evict immediately)
+                      by the system.
+                    format: int64
```
[the remark this answers is not in the document]
Comment (jsanda) [reply config/crd/bases/reaper.cassandra-reaper.io_reapers.yaml:127]: Same as above. This is part of the core Toleration type.

**Pull request merged**